src/dbnode/storage/database.go

```diff
@@ -694,17 +695,22 @@ func (d *db) writeBatch(
 			errHandler.HandleError(write.OriginalIndex, err)
 		}
 
-		// Need to set the outcome in the success case so the commitlog gets the updated
-		// series object which contains identifiers (like the series ID) whose lifecycle
-		// live longer than the span of this request, making them safe for use by the async
-		// commitlog. Need to set the outcome in the error case so that the commitlog knows
-		// to skip this entry.
-		writes.SetOutcome(i, series, err)
+		if !wasWritten && err == nil {
+			// This series has no additional information that needs to be written to
+			// the commit log; set this series to skip writing to the commit log.
+			writes.SetSkipWrite(i)
+		} else {
+			// Need to set the outcome in the success case so the commitlog gets the
+			// updated series object which contains identifiers (like the series ID)
+			// whose lifecycle lives longer than the span of this request, making them
+			// safe for use by the async commitlog. Need to set the outcome in the
+			// error case so that the commitlog knows to skip this entry.
+			writes.SetOutcome(i, series, err)
```

Comment:
Not sure if I brought this up last time, but is there any reason we can't just `SetOutcome` in both cases? Feels a little safer to make sure everything gets updated anyways and I'm pretty sure its cheap since it literally just does this

```
func (b *writeBatch) SetOutcome(idx int, series Series, err error) {
	b.writes[idx].Write.Series = series
	b.writes[idx].Err = err
}
```

Reply:
So basically always SetOutcome

and then 

```golang
if !wasWritten || err != nil {
    writes.SetSkipWrite(i)
}
```

Reply:
Notice that I changed it to an OR because you always want to skip the write if it wasn't written or there was an error. I know the commitlog will skip it anyways if there was an error but this seems more semantically correct

Reply:
Sure, sounds good to me; makes sense that the readability is worth more here